app/services/offender_service.rb

```diff
@@ -5,8 +5,8 @@ def self.get_offender(offender_no)
     HmppsApi::PrisonApi::OffenderApi.get_offender(offender_no).tap { |o|
       next false if o.nil?
 
-      record = CaseInformation.find_by(nomis_offender_id: offender_no)
-      o.load_case_information(record)
+      offender = Offender.find_or_create_by!(nomis_offender_id: offender_no)
+      o.load_case_information(offender.case_information)
```

Comment:
Shouldn't we be doing the same in the `prison.offenders` method?

The method `Prison::OffenderEnumerator#enrich_offenders` currently uses `CaseInformationService` to get offenders. But it doesn't auto-create offender records – meaning 'full fat' offender objects will probably end up without case information being loaded.